lib/heex_formatter/phases/format.ex

```diff
@@ -85,9 +83,23 @@ defmodule HeexFormatter.Phases.Format do
         end
       end
 
+    mode = if tag in ~w(script tag style code pre), do: String.to_atom(tag), else: :normal
     indentation = if self_closed?, do: state.indentation, else: state.indentation + 1
 
-    %{state | html: state.html <> "\n" <> tag_opened, indentation: indentation}
+    %{state | html: state.html <> "\n" <> tag_opened, indentation: indentation, mode: mode}
+  end
+
+  defp token_to_string({tag, text, _meta}, %{mode: mode} = state)
+       when tag in ~w(text eex_tag eex_tag_render)a and mode in ~w(script style code pre comment)a do
+    %{state | html: state.html <> text}
+  end
+
+  defp token_to_string({:text, text, %{context: :comment_start}}, state) do
+    %{state | html: state.html <> String.trim_trailing(text), mode: :comment}
+  end
+
+  defp token_to_string({:text, text, %{context: :comment_end}}, state) do
+    %{state | html: state.html <> String.trim_trailing(text), mode: :normal}
   end
```

Comment:
@josevalim The current code introduced in https://github.com/phoenixframework/phoenix_live_view/pull/1847 is enough to achieve what want here. It turns out that only adding `comment_start` and `comment_end` is enough since the script, among other tags we want to ignore, we handle by the tag itself because we know when it ends.

So now, we would have only two interactions here:
1. To build our tokens combining both `EExTonenizer` and `HTMLTokenizer`.
2. For formatting the code based on the tokens.

We might be able to format in just one go, but I guess it would make the code a bit hard to follow 🤔 

Please, let me know if that make sense 🙏🏻